Objective-C/CBLDatabase.mm

```diff
@@ -193,12 +217,9 @@ - (BOOL) purgeDocument: (CBLDocument*)document error: (NSError**)error {
         if (![self prepareDocument: document error: error])
             return NO;
         
-        if (!document.revID) {
-            [NSException raise: NSInvalidArgumentException
-                        format: @"Do not allow to purge a newly created document "
-                                 "that has not been saved into the database."];
-            return NO;
-        }
+        if (!document.revID)
+            return createError(CBLErrorInvalidParameter,
```

Comment:
Shouldn't this be CBLErrorNotFound?

Reply:
It's 50/50 between CBLErrorInvalidParameter or CBLErrorNotFound. For example, the document is actually in the database but we don't allow this because the document should be gotten from the database first.

Reply:
I created https://github.com/couchbase/couchbase-lite-ios/issues/2075 to track this and get opinion from @borrrden .